.github/workflows/go.yml

```diff
@@ -0,0 +1,202 @@
+name: CI
+
+on:
+  push:
+    branches:
+      - main
+      - release/*
+    tags:
+      - v[0-9]+.[0-9]+.[0-9]+*
+  pull_request:
+
+jobs:
+  check:
+    runs-on: ubuntu-20.04
+    strategy:
+      matrix:
+        go:
+          - version: 1.15.x
+            may-fail: false
+          - version: 1.16.x
+            may-fail: false
+          # TODO enable when 1.15.x is removed
+#          - version: tip
+#            may-fail: true
+
+    continue-on-error: ${{ matrix.go.may-fail }}
+
+    env:
+      GOPATH: ${{ github.workspace }}
+      WORKING_DIR: ${{ github.workspace }}/src/github.com/percona/pmm-agent
+      GO111MODULE: off #TODO: remove once we move to go modules https://jira.percona.com/browse/PMM-7106
+
+    defaults:
+      run:
+        working-directory: ${{ env.WORKING_DIR }}
+
+    steps:
+      - name: Check out code into the working directory under the GOPATH
+        uses: percona-platform/checkout@v2
+        with:
+          fetch-depth: 1
+          path: ${{ env.WORKING_DIR }}
+
+      - name: Set up Go release
+        if: matrix.go.version != 'tip'
+        uses: percona-platform/setup-go@v2
+        with:
+          go-version: ${{ matrix.go.version }}
+
+      - name: Set up Go tip
+        if: matrix.go.version == 'tip'
+        run: |
+          git clone --depth=1 https://go.googlesource.com/go $HOME/gotip
+          cd $HOME/gotip/src
+          ./make.bash
+          echo "GOROOT=$HOME/gotip" >> $GITHUB_ENV
+          echo "$HOME/gotip/bin" >> $GITHUB_PATH
+
+      - name: go env
+        run: go env
```

Comment:
Propose to replace it with:
```
      - name: Run debug commands on failure
        if: ${{ failure() }}
        run: |
          env
          go version
          go env
          pwd
          git status
```
and set it as the last step

Reply:
Done